coffee/base.py

```diff
@@ -139,9 +147,11 @@ class BinExpr(Expr):
 
     """Generic binary expression."""
 
-    def __init__(self, expr1, expr2, op):
+    def __init__(self, expr1, expr2):
         super(BinExpr, self).__init__([expr1, expr2])
-        self.op = op
+
+    def reconstruct(self, expr1, expr2, **kwargs):
```

Comment:
So all reconstruct methods have kwargs, I was mimicking the previous code for init